rollup/internal/controller/sender/sender.go

```diff
@@ -159,7 +159,16 @@ func (s *Sender) SendConfirmation(cfm *Confirmation) {
 func (s *Sender) getFeeData(target *common.Address, data []byte, sidecar *gethTypes.BlobTxSidecar, baseFee, blobBaseFee uint64, fallbackGasLimit uint64) (*FeeData, error) {
 	switch s.config.TxType {
 	case LegacyTxType:
-		return s.estimateLegacyGas(target, data, fallbackGasLimit)
+		feeData, err := s.estimateLegacyGas(target, data, fallbackGasLimit)
+		if err != nil {
+			return nil, err
+		}
+		baseFeeInt := new(big.Int).SetUint64(baseFee)
+		maxGasPrice := new(big.Int).SetUint64(s.config.MaxGasPrice)
+		if feeData.gasPrice.Cmp(baseFeeInt) < 0 && baseFeeInt.Cmp(maxGasPrice) <= 0 {
+			feeData.gasPrice = baseFeeInt
```

Comment:
this is updated by `gasPrice, err := s.client.SuggestGasPrice(s.ctx)` in `estimateLegacyGas`.

Reply:
you mean 
```
func (s *EthereumAPI) GasPrice(ctx context.Context) (*hexutil.Big, error) {
	tipcap, err := s.b.SuggestGasTipCap(ctx)
	if err != nil {
		return nil, err
	}
	if head := s.b.CurrentHeader(); head.BaseFee != nil {
		tipcap.Add(tipcap, head.BaseFee)
	}
	return (*hexutil.Big)(tipcap), err
}
```

The weird thing is that we might still get suggeted gas price under basefee. See PR description and altlayer's feedback.

Reply:
maybe the block's base fee is updated after the txn is submitted.

Reply:
reverted in 30a7b47b